ee/tuf/autoupdate.go

```diff
@@ -709,6 +726,32 @@ func findRelease(ctx context.Context, binary autoupdatableBinary, targets data.T
 	return "", data.TargetFileMeta{}, fmt.Errorf("could not find metadata for release target %s for binary %s", releaseTarget, binary)
 }
 
+// findReleasePromoteTime extracts the promotion timestamp from the release file metadata for a given binary and channel.
+// It searches the TUF targets for the appropriate release.json file based on the binary, OS, architecture, and channel,
+// then unmarshals the custom metadata to retrieve the PromoteTime field.
+// Returns the Unix timestamp of when the release was promoted, or 0 if the release file is not found or cannot be parsed.
+func findReleasePromoteTime(ctx context.Context, binary autoupdatableBinary, targets data.TargetFiles, channel string) int64 {
```

Comment:
Why do we have to find the release metadata and unmarshall it separately here, instead of doing this in `findRelease`?

Reply:
it seemed like `findRelease` was returning the `TargetFileMeta` associated with the target itself (just using the release.json entry to find that). but the new metadata is stored in the release.json entries custom blob- I suppose we could take the promote time while unmarshalling for the custom `Target` there and add it to the actual TargetFileMeta.Custom returned, but that didn't really seem right to me either